src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -172,7 +172,7 @@ public void AddContext<TContext>() where TContext : JsonSerializerContext, new()
         {
             VerifyMutable();
             TContext context = new();
-            context.Options = this;
+            TypeInfoResolver = JsonTypeInfoResolver.Combine(TypeInfoResolver, context);
```

Comment:
I'd recommend to allow that only if CombinedContext consists only of other contexts - I don't feel super strongly about it but I'd prefer we block everything w.r.t AddContext which won't end up (or rather can't end up) with working fast path

Reply:
I've updated the `Combine` method so that if the arguments are also combine contexts, then their arguments are flattened to a single resolver.

Reply:
That makes sense and I agree we should do that. I think AddContext should throw though if TypeInfoResolver was set to something else than context/combined context i.e. DefaultJsonTypeInfoResolver/Custom resolver.

Reply:
> I think AddContext should throw though if TypeInfoResolver was set to something else than context/combined context i.e. DefaultJsonTypeInfoResolver/Custom resolver.

I think that might prevent use cases that are valid, e.g.
```C#
options.TypeInfoResolver = new CustomResolverWithModifiersOverSomeContext();
/* later, in some other configuration method */
options.AddContext<ContextB>();
```